Dockerfiles/pcluster-amazonlinux-2/Dockerfile

```diff
@@ -141,6 +136,28 @@ RUN --mount=type=secret,id=bootstrap_gcc_key \
     && spack gpg publish --rebuild-index -m bootstrap-gcc-cache ${secretkey_fingerprint} \
     && rm -rf $(dirname "${SPACK_ROOT}") /root/.spack
 
+# Add oneapi compiler into container. The stack can pick up the compilers by using ${SPACK_ROOT}/etc/spack/compilers.yaml
+# intel-oneapi-compiler@2024.1.0 is the last version compatible with AL2 glibc 2.26
+COPY Dockerfiles/pcluster-amazonlinux-2/${TARGETARCH}/packages.yaml /root/.spack/packages.yaml
+RUN if [[ "amd64" == "${TARGETARCH}" ]]; then mkdir -p $(dirname "${SPACK_ROOT}") \
+    && git clone --depth=1 -b ${SPACK_TAG} https://github.com/spack/spack.git "${SPACK_ROOT}" \
+    && cp "${SPACK_ROOT}/share/spack/gitlab/cloud_pipelines/configs/config.yaml" "${SPACK_ROOT}/etc/spack/config.yaml" \
+    && cp "${SPACK_ROOT}/share/spack/gitlab/cloud_pipelines/configs/config.yaml" "/bootstrap/cloud_pipelines-config.yaml" \
+    && . "${SPACK_ROOT}/share/spack/setup-env.sh" \
+    && spack load gcc \
+    && spack compiler add --scope site \
+    && cd "${SPACK_ROOT}" \
+    && spack install intel-oneapi-compilers@2024.1.0 \
+    && . "$(spack location -i intel-oneapi-compilers)"/setvars.sh \
+    && spack compiler add --scope site \
```

Comment:
https://spack.readthedocs.io/en/latest/build_systems/inteloneapipackage.html suggests this line should be more like
```
spack compiler add `spack location -i intel-oneapi-compilers`/compiler/latest/bin --scope site \
```